sympy/matrices/expressions/matpow.py

```diff
@@ -4,18 +4,19 @@
 from sympy.core.sympify import _sympify
 from sympy.core.compatibility import range
 from sympy.matrices import MatrixBase
-from sympy.core import S
+from sympy.core import S, Symbol
 
 
 class MatPow(MatrixExpr):
 
-    def __new__(cls, base, exp):
+    def __new__(cls, base, exp, evaluate=False):
```

Comment:
> MatPow is an exception.The normal behaviour of MatPow is to be unevaluated(that is actually the purpose of it).

Maybe it should remain unevaluated if the matrix power algorithm fails. I don't see the problem in evaluating it.

> Would it be desirable behaviour to have it return X**2 in the evaluate(False) block ?

Yes, in an `evaluate(False)` scope, `X**2` should remain unevaluated.

In other cases, my opinion is still that `X**2` should be evaluated, while `MatPow(X, 2, evaluate=False)` shall return the unevaluated form.